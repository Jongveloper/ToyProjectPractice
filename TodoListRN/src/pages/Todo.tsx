import React from 'react';
import { Text, View } from 'react-native';

function SignUp() {
  return (
    <View>
      <Text>투두리스트</Text>
    </View>
  )
}

export default SignUp;