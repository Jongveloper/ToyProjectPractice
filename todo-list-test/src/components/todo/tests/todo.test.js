import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Todo from '../todo';

describe('Todo component', () => {
  const todo = { name: 'Learn React' };
  let onDelete;

  beforeEach(() => {
    onDelete = jest.fn();
  });

  it('renders', () => {
    const component = renderer.create(<Todo todo={todo} onDelete={onDelete} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe('Button Click', () => {
    const setup = () => render(<Todo todo={todo} onDelete={onDelete} />);

    it('투두리스트를 삭제합니다.', () => {
      setup();
      const button = screen.getByTitle('삭제');
      userEvent.click(button);
      expect(onDelete).toHaveBeenCalledWith(todo);
    });
  });
});
