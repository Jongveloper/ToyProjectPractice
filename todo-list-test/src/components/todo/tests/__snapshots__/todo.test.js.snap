// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component renders 1`] = `
<li>
  <span />
  <button
    onClick={[Function]}
    title="삭제"
  >
    <i
      className="fas fa-trash"
    />
  </button>
</li>
`;
